gravitee-am-test/specs/gateway/am-selfaccount-change-password.jest.spec.ts

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import fetch from "cross-fetch";
+
+global.fetch = fetch;
+
+import {jest, afterAll, beforeAll, expect} from "@jest/globals";
+import {requestAdminAccessToken} from "@management-commands/token-management-commands";
+import {createDomain, deleteDomain, patchDomain, startDomain} from "@management-commands/domain-management-commands";
+import {getAllIdps} from "@management-commands/idp-management-commands";
+import {createUser} from "@management-commands/user-management-commands";
+import {
+    createApplication, 
+    updateApplication
+} from "@management-commands/application-management-commands";
+import {
+    getWellKnownOpenIdConfiguration,
+    performPost
+} from "@gateway-commands/oauth-oidc-commands";
+import {applicationBase64Token} from "@gateway-commands/utils";
+
+let domain;
+let managementApiAccessToken;
+let endUserAccessToken;
+let openIdConfiguration;
+let application;
+let user = {
+    "username": "SelfAccountUser",
+    "password": "Test123!",
+    "firstName": "SelfAccount",
+    "lastName": "User",
+    "email": "selfaccountuser@acme.fr",
+    "preRegistration": false
+}
+const PATCH_DOMAIN_SEFLACCOUNT_SETTINGS = {
+    "selfServiceAccountManagementSettings": { 
+        "enabled":true, 
+        "resetPassword": {
+            "oldPasswordRequired":false,
+            "tokenAge":0
+        }
+    }
+}
+
+jest.setTimeout(200000)
+
+const waitForDomainSync = () => new Promise((r) => setTimeout(r, 10000))
+
+beforeAll(async () => {
+    const adminTokenResponse = await requestAdminAccessToken();
+    managementApiAccessToken = adminTokenResponse.body.access_token;
+    expect(managementApiAccessToken).toBeDefined();
+    const createdDomain = await createDomain(managementApiAccessToken, "self-account-change-password", "test Change Password through SelfAccount API");
+    expect(createdDomain).toBeDefined();
```

Comment:
this expect is not necessary. The next line will cover both

Reply:
I'm not agree, this line test that the REST API return an object / content.
The next one test that the content has an ID. If we do not test the "createDomain" definition, we will have an error due to undefined object so it will be a bug into the test. Result will be the same, as the test will fail but the semantic is different.

Reply:
Your view is correct but overall it is not necessary to have assertion in setup stage all . If error is thrown then it is an error which not relates to the test. But all the practices are common: Here is a link worth reading: https://stackoverflow.com/questions/1371940/unit-testing-is-it-a-good-practice-to-have-assertions-in-setup-methods